nengo/cache.py

```diff
@@ -648,15 +647,8 @@ def cached_solver(conn, gain, bias, x, targets,
                 return solver_fn(conn, gain, bias, x, targets,
                                  rng=rng, E=E, **uncached_kwargs)
 
-            try:
-                args, _, _, defaults = inspect.getargspec(conn.solver)
-            except TypeError:
-                args, _, _, defaults = inspect.getargspec(conn.solver.__call__)
-            args = args[-len(defaults):]
-            if rng is None and 'rng' in args:
-                rng = defaults[args.index('rng')]
-            if E is None and 'E' in args:
-                E = defaults[args.index('E')]
+            if rng is None:
+                rng = np.random
```

Comment:
Can I ask if there's a reason to specify the default this way rather than in the arg list? I know putting mutable things as a default can be problematic, but in this case `np.random` is mutable but also global so it would be the same were it mutated.

Reply:
I am just following what's done in [`solvers.py`](https://github.com/nengo/nengo/blob/master/nengo/solvers.py), but I was wondering the same thing.

Reply:
I say change it both places :+1:

Reply:
Can do! Tagging @hunse who changed it to what it is now in 63c21040fd7f5ac139a18efe552ef882d44a5f59 in case we overlooked something.

Reply:
Why did I change this... that's a good question! It might have been so that all the solvers would have the same default ``rng=None``, whether they actually use an ``rng`` or not, and maybe I thought it kept the function signature a bit cleaner, too. But I'm fine making the change suggested here.

Reply:
Changed in both places. :)